File: matiasmasso/Mat.NET/DT2/DTOHolding.vb
Public Class DTOHolding
    Inherits DTOGuidNom
    Property Emp As DTOEmp
    Property Companies As List(Of DTOContact)
    Property Clusters As List(Of DTOCustomerCluster)

    Public Enum wellknowns
        NotSet
        ElCorteIngles
        Prenatal
    End Enum

    Public Sub New()
        MyBase.New
        _Companies = New List(Of DTOContact)
        _Clusters = New List(Of DTOCustomerCluster)
    End Sub

    Public Sub New(oGuid As Guid)
        MyBase.New(oGuid)
        _Companies = New List(Of DTOContact)
        _Clusters = New List(Of DTOCustomerCluster)
    End Sub

    Overloads Shared Function Factory(oEmp As DTOEmp, sNom As String) As DTOHolding
        Dim retval As New DTOHolding
        With retval
            .Emp = oEmp
            .nom = sNom
        End With
        Return retval
    End Function

    Shared Function wellknown(id As wellknowns) As DTOHolding
        Dim retval As DTOHolding = Nothing
        Select Case id
            Case wellknowns.ElCorteIngles
                retval = New DTOHolding(New Guid("8B07C540-1DA6-48E2-B137-563BFDD4218B"))
            Case wellknowns.Prenatal
                retval = New DTOHolding(New Guid("1908D0F6-4B81-464F-B5CC-BE1B70AC8B3B"))
        End Select
        Return retval
    End Function
End Class

File: matiasmasso/Mat.NET/DTO/DTOHolding.cs
using System;
using System.Collections.Generic;

namespace DTO
{
    public class DTOHolding : DTOGuidNom
    {
        public DTOEmp emp { get; set; }
        public List<DTOContact> companies { get; set; }
        public List<DTOCustomerCluster> clusters { get; set; }

        public enum Wellknowns
        {
            notSet,
            elCorteIngles,
            prenatal
        }

        public DTOHolding() : base()
        {
            companies = new List<DTOContact>();
            clusters = new List<DTOCustomerCluster>();
        }

        public DTOHolding(Guid oGuid) : base(oGuid)
        {
            companies = new List<DTOContact>();
            clusters = new List<DTOCustomerCluster>();
        }

        public static DTOHolding Factory(DTOEmp oEmp, string sNom)
        {
            DTOHolding retval = new DTOHolding();
            {
                var withBlock = retval;
                withBlock.emp = oEmp;
                withBlock.Nom = sNom;
            }
            return retval;
        }

        public static DTOHolding Wellknown(Wellknowns id)
        {
            DTOHolding retval = null;
            switch (id)
            {
                case Wellknowns.elCorteIngles:
                    {
                        retval = new DTOHolding(new Guid("8B07C540-1DA6-48E2-B137-563BFDD4218B"));
                        break;
                    }

                case Wellknowns.prenatal:
                    {
                        retval = new DTOHolding(new Guid("1908D0F6-4B81-464F-B5CC-BE1B70AC8B3B"));
                        break;
                    }
            }
            return retval;
        }
    }
}
